### PR TITLE
Add options for hcaptcha and cachebuster

### PR DESCRIPTION
**Pull request by timetinytim**

### CODE REVIEW

File: templates/configmap-env.yaml [hunk 1]
```diff
@@ -344,3 +344,22 @@ data:
...
+  {{- else }}
+  CACHE_BUSTER_ENABLED: "false"
+  {{- end }}
+  {{- if .Values.mastodon.cacheBuster.httpMethod }}
+  CACHE_BUSTER_HTTP_METHOD: {{ .Values.mastodon.cacheBuster.httpMethod }}
+  {{- end }}
+  {{- if .Values.mastodon.cacheBuster.authHeader }}
+  CACHE_BUSTER_SECRET_HEADER: {{ .Values.mastodon.cacheBuster.authHeader }}
+  {{- end }}
+  {{- range $k, $v := .Values.mastodon.extraEnvVars }}
```
Comment (renchap) [inline templates/configmap-env.yaml:363]: You forgot to remove this above, so those lines are now duplicated.

We need them at the end of the file, probably with a new line before to separate them

File: templates/configmap-env.yaml [hunk 2]
```diff
@@ -344,3 +344,22 @@ data:
...
+  {{- if .Values.mastodon.hcaptcha.enabled }}
+  HCAPTCHA_SITE_KEY: {{ .Values.mastodon.hcaptcha.siteId }}
+  {{- end }}
+
+  {{- if .Values.mastodon.cacheBuster.enabled }}
+  CACHE_BUSTER_ENABLED: "true"
+  {{- else }}
+  CACHE_BUSTER_ENABLED: "false"
+  {{- end }}
+  {{- if .Values.mastodon.cacheBuster.httpMethod }}
```
Comment (renchap) [inline templates/configmap-env.yaml:357]: Maybe only add the values below if the cache buster is enabled?

File: values.yaml [hunk 3]
```diff
@@ -211,6 +217,16 @@ mastodon:
     workers: "2"
     persistentTimeout: "20"
 
+  # Additional settings for the cache buster.
+  cacheBuster:
```
Comment (renchap) [inline values.yaml:221]: We may want a link to the documentation for this feature, as the format is specific

File: templates/deployment-web.yaml [hunk 4]
```diff
@@ -123,6 +123,20 @@ spec:
...
                   key: {{ .Values.mastodon.deepl.apiKeySecretRef.key }}
             {{- end }}
+            {{- if .Values.mastodon.hcaptcha.enabled }}
+            - name: "HCAPTCHA_SECRET_KEY"
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.mastodon.hcaptcha.secretKeySecretRef.name }}
+                  key: {{ .Values.mastodon.hcaptcha.secretKeySecretRef.key }}
+            {{- end }}
+            {{- if .Values.mastodon.cacheBuster.authToken.existingSecret }}
```
Comment (renchap) [inline templates/deployment-web.yaml:133]: Only if enabled?

File: templates/deployment-sidekiq.yaml [hunk 5]
```diff
@@ -135,6 +135,13 @@ spec:
                   name: {{ $context.Values.mastodon.deepl.apiKeySecretRef.name }}
                   key: {{ $context.Values.mastodon.deepl.apiKeySecretRef.key }}
             {{- end }}
+            {{- if $context.Values.mastodon.cacheBuster.authToken.existingSecret }}
```
Comment (renchap) [inline templates/deployment-sidekiq.yaml:138]: Only if enabled?

**Review by renchap — CHANGES_REQUESTED** [review]

File: templates/configmap-env.yaml (hunk 1 above)
Comment (timetinytim) [reply templates/configmap-env.yaml:363]: Oh good catch, that's my bad

File: values.yaml (hunk 3 above)
Comment (timetinytim) [reply values.yaml:221]: Where is the documentation for this stuff specifically? I looked myself and couldn't find more details about mastodon's cache buster specifically.

Comment (renchap) [reply values.yaml:221]: We do not seem to have a documentation on this…

Comment (renchap) [reply values.yaml:221]: I added documentation in this PR: https://github.com/mastodon/documentation/pull/1383

Once it is merge, can you update this to include a link to the doc?

Comment (timetinytim) [reply values.yaml:223]: Awesome! Added links in the comments.

**Review by renchap — APPROVED** [review]

**Pull request merged**